modules/invenio-oaiserver/invenio_oaiserver/response.py

```diff
@@ -400,23 +400,28 @@ def listrecords(**kwargs):
         return error(get_error_code_msg(), **kwargs)
 
     for record in result.items:
-        pid = oaiid_fetcher(record['id'], record['json']['_source'])
-        e_record = SubElement(e_listrecords,
-                              etree.QName(NS_OAIPMH, 'record'))
-        header(
-            e_record,
-            identifier=pid.pid_value,
-            datestamp=record['updated'],
-            sets=record['json']['_source'].get('_oai', {}).get('sets', []),
-        )
-        from weko_deposit.api import WekoRecord
-        db_record = WekoRecord.get_record(record['id'])
-        if not record['json']['_source']['_item_metadata']\
-                .get('system_identifier_doi'):
-            record['json']['_source']['_item_metadata'][
-                'system_identifier_doi'] = get_identifier(db_record)
-        e_metadata = SubElement(e_record, etree.QName(NS_OAIPMH, 'metadata'))
-        e_metadata.append(record_dumper(pid, record['json']))
+        try:
+            pid = oaiid_fetcher(record['id'], record['json']['_source'])
+            e_record = SubElement(e_listrecords,
+                                  etree.QName(NS_OAIPMH, 'record'))
+            header(
+                e_record,
+                identifier=pid.pid_value,
+                datestamp=record['updated'],
+                sets=record['json']['_source'].get('_oai', {}).get('sets', []),
+            )
+            from weko_deposit.api import WekoRecord
+            db_record = WekoRecord.get_record(record['id'])
+            if not record['json']['_source']['_item_metadata']\
+                    .get('system_identifier_doi'):
+                record['json']['_source']['_item_metadata'][
+                    'system_identifier_doi'] = get_identifier(db_record)
+            e_metadata = SubElement(e_record, etree.QName(NS_OAIPMH, 'metadata'))
+            e_metadata.append(record_dumper(pid, record['json']))
+        except Exception:
+            import traceback
```

Comment:
maybe traceback unnecessary
can:
exception Exception as exc:
    print(exc)

Reply:
Actually, function above calls many deep other functions, so `exc` does not return a specific traceback but a common message instead, an uninformative msg like that:
`index out of range`
So that I had to use traceback() in order to trace the detailed error.